indexer/search_string_utils.cpp

```diff
@@ -94,11 +96,67 @@ char const * kStreetTokensSeparator = "\t -,.";
 
 /// @todo Move prefixes, suffixes into separate file (autogenerated).
 /// It's better to distinguish synonyms comparison according to language/region.
-
 class StreetsSynonymsHolder
 {
 public:
-  using TStrings = search::StringSet<UniChar, 8>;
+  struct BooleanSum
+  {
+    void Add(bool value) { m_value = m_value || value; }
+
+    template <typename ToDo>
+    void ForEach(ToDo && toDo) const
+    {
+      if (m_value)
+        toDo(m_value);
```

Comment:
Moreover, without value you won't be able to distinguish between internal nodes and nodes that mean end of some string.